packages/zwave-js/src/lib/driver/Driver.ts

```diff
@@ -2412,6 +2413,53 @@ ${handlers.length} left`,
 				if (!msg.command.moreUpdatesFollow) {
 					nodeSessions.supervision.delete(msg.command.sessionId);
 				}
+			} else if (
+				msg.command.encapsulatingCC instanceof SupervisionCCGet
+			) {
+				// check if someone is waiting for this command
+				for (const entry of this.awaitedCommands) {
+					if (entry.predicate(msg.command)) {
+						// resolve the promise - this will remove the entry from the list
+						entry.promise.resolve(msg.command);
+						return;
+					}
+				}
+
+				if (msg.command instanceof InvalidCC) {
+					const report = new SupervisionCCReport(this, {
+						sessionId: msg.command.encapsulatingCC.sessionId,
+						moreUpdatesFollow: false,
+						nodeId: msg.command.encapsulatingCC.nodeId,
+						status: SupervisionStatus.Fail,
+						duration: new Duration(0, "seconds"),
+					});
+
+					// The report should be sent back with security if the received command was secure
+					report.secure = msg.command.secure;
+
+					await this.sendCommand(report, {
+						priority: MessagePriority.Handshake,
+					});
+
+					return;
+				}
+
+				await node.handleCommand(msg.command);
+
+				const report = new SupervisionCCReport(this, {
+					sessionId: msg.command.encapsulatingCC.sessionId,
+					moreUpdatesFollow: false,
+					nodeId: msg.command.encapsulatingCC.nodeId,
+					status: SupervisionStatus.Success,
+					duration: new Duration(0, "seconds"),
+				});
```

Comment:
A nice extra would be to set the "wake up request" bit if we know that the device supports Supervision CC version 2, WakeUp, and there are pending commands to be sent.

Probably this should be done into a separate PR, but I just wanted to note it.

Reply:
That would solve part (if not all) of https://github.com/zwave-js/node-zwave-js/issues/1158
Should be done in a separate PR.